### PR TITLE
Explainer: section on errors, start on device loss

### PR DESCRIPTION
**Pull request by kainino0x**
Heavily cribbed from [ErrorHandling.md](https://github.com/gpuweb/gpuweb/blob/main/design/ErrorHandling.md) but updates some things and omits a bunch.

### CODE REVIEW

File: explainer/index.bs [hunk 1]
```diff
@@ -170,22 +174,246 @@ Of course, it becomes a validation to use the object after the call to `.destroy
...
+the first error that was captured by the scope - either a `GPUValidationError` or a
+`GPUOutOfMemoryError`.
+
+An error scope captures an error if its filter (`'validation'` or `'out-of-memory'`) matches the
+type of the error.
+This filter mechanism avoids, e.g., accidentally silencing validation errors when trying to
+implement fallible allocation.
+
+Any device-timeline error from an operation is passed to the top-most error scope on the stack at
+the time it was issued.
```
Comment (kainino0x) [inline explainer/index.bs:252]: @Kangz:
> Maybe explain that this we done for improved composability.

Not sure what you meant by this, could you explain a bit more?

Comment (Kangz) [reply explainer/index.bs:255]: We should describe the reason we have a stack of error scopes and not just a single-level error scope (or something like async glGetError). It is to improve the composability of the application. For example a middleware can choose to catch OOM errors on an allocation while the rest of the application doesn't know about it. It makes the error API more stateless by making correctly written code (that balance error scopes) blind to whether it is already in error scopes or not.

**Review by Kangz — APPROVED** [review]

**Review by kvark — APPROVED** [review]
Reads very well, good job!

**Comment by kainino0x** [comment]
Thanks! 😄

**Pull request merged**